### PR TITLE
Remove the lightbeam backend

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit removes the Lightbeam backend from Wasmtime as per [RFC 14].
This backend hasn't received maintenance in quite some time, and as [RFC
14] indicates this doesn't meet the threshold for keeping the code
in-tree, so this commit removes it.

A fast "baseline" compiler may still be added in the future. The
addition of such a backend should be in line with [RFC 14], though, with
the principles we now have for stable releases of Wasmtime. I'll close
out Lightbeam-related issues once this is merged.

[RFC 14]: https://github.com/bytecodealliance/rfcs/pull/14

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -155,10 +146,6 @@ struct CommonOptions {
     #[structopt(long, parse(from_os_str), value_name = "CONFIG_PATH")]
     config: Option<PathBuf>,
 
-    /// Use Cranelift for all compilation
-    #[structopt(long, conflicts_with = "lightbeam")]
-    cranelift: bool,
```
Comment (bjorn3) [inline src/lib.rs:160]: Maybe keep this for in the future if a new backend is added?

**Review by bjorn3 — APPROVED** [review]

File: crates/c-api/include/wasmtime/config.h [hunk 2]
```diff
@@ -26,19 +26,13 @@ typedef uint8_t wasmtime_strategy_t;
...
  */
 enum wasmtime_strategy_enum { // Strategy
-  /// Wasmtime will automatically determine whether to use Cranelift or
-  /// Lightbeam, and currently it will always pick Cranelift. This default may
-  /// change over time though.
+  /// Automatically picks the compilation backend, currently always defaulting
+  /// to Cranelift.
   WASMTIME_STRATEGY_AUTO,
 
   /// Indicates that Cranelift will unconditionally use Cranelift to compile
```
Comment (cfallin) [inline crates/c-api/include/wasmtime/config.h:33]: Pre-existing, but s/Cranelift will unconditionally use Cranelift/Wasmtime will unconditionally use Cranelift/ ?

File: docs/contributing-architecture.md [hunk 3]
```diff
@@ -120,11 +120,9 @@ their implications in Wasmtime:
 * `wasmtime::Module` - this is the representation of a compiled WebAssembly
   module. At this time Wasmtime always assumes that a wasm module is always
   compiled to native JIT code. `Module` holds the results of said compilation,
-  and currently the Cranelift and Lightbeam modes can be used for compiling. It
-  is a goal of Wasmtime to support other modes of representing modules but those
-  are not implemented today just yet. Additionally the Lightbeam compiler has
-  not received maintenance in a long time, so effectively everyone uses
-  Cranelift.
+  and currently the Cranelift can be used for compiling. It is a goal of
```
Comment (cfallin) [inline docs/contributing-architecture.md:123]: s/the Cranelift/Cranelift/

**Review by cfallin — APPROVED** [review]
LGTM!

File: src/lib.rs (hunk 1 above)
Comment (alexcrichton) [reply src/lib.rs:160]: This can be added in the future and otherwise I think it would be confusing to have a flag that basically does nothing.

**Pull request merged**